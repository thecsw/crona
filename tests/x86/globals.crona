i : int;
by : byte;
b : bool;

set_i : void (new_i : int) {
    i = new_i;
}

set_by : void (new_by : byte) {
    by = new_by;
}

set_b : void (new_b : bool) {
    b = new_b;
}

main : void() {
    i = -500;
    write "old i=";
    write i;
    write "; new i=";
    set_i(500);
    write i;
    write "\n";

    by = 20;
    write "old by=";
    write by;
    write "; new by=";
    set_by(255);
    write by;
    write "\n";

    b = false;
    write "old b=";
    write b;
    write "; new b=";
    set_b(true);
    write b;
    write "\n";
}