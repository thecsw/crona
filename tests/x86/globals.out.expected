old i=-500; new i=500
old by=20; new by=255
old b=false; new b=true
